bundles/com.zeligsoft.cx.codegen.ui/src/com/zeligsoft/cx/codegen/ui/preferences/CodegenReportPreferencePage.java

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright 2021 Zeligsoft.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.zeligsoft.cx.codegen.ui.preferences;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.runtime.preferences.ConfigurationScope;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.IPreferencesService;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.jface.preference.PreferencePage;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.ControlAdapter;
+import org.eclipse.swt.events.ControlEvent;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Group;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.ui.IWorkbench;
+import org.eclipse.ui.IWorkbenchPreferencePage;
+import org.osgi.service.prefs.BackingStoreException;
+
+import com.zeligsoft.base.Activator;
+import com.zeligsoft.cx.CXActivator;
+import com.zeligsoft.cx.preferences.CXPreferenceConstants;
+import com.zeligsoft.cx.codegen.ui.l10n.Messages;
+
+/**
+ * CX codegen report preference page.
+ * 
+ * @author Ernesto Posse
+ */
+public class CodegenReportPreferencePage extends PreferencePage implements IWorkbenchPreferencePage {
+
+	private static final IPreferencesService PREFERENCE_SERVICE = Platform.getPreferencesService();
+
+	private static final IEclipsePreferences WORKSPACE_PREFERENCES = InstanceScope.INSTANCE
+			.getNode(CXActivator.PLUGIN_ID);
```

Comment:
Do we need to use the cx activator for store? Can we use more specific bundle? like codegen.ui.Activator?